ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/action/ActionManager.java

```diff
@@ -114,4 +115,16 @@
      * @see PromisableAction
      */
     Promise<Void> performActions(List<Pair<Action, ActionEvent>> actions, boolean breakOnFail);
+
+    /**
+     * Performs registered action by given id.
+     *
+     * <p>Note that if action is not registered nothing will be done
+     *
+     * @param actionId
+     *         the id of action that will be performed
+     * @param parameters
+     *         the parameters with which the action is invoked or updated
+     */
+    void performAction(String actionId, Map<String, String> parameters);
```

Comment:
What if `actionId` or `parameters` map is null ? How should an implementation behave in this case.
It is always nice to mark parameters with `@Nullable` and `@NotNull` annotations + define behaviour.